vaxrank/vaccine_peptide.py

```diff
@@ -54,14 +60,35 @@ def lexicographic_sort_key(self):
         score positive (since we want fewer wildtype epitopes) but the others
         negative (since we want more of them).
         """
-        return (
+        # since we're sorting in decreasing order, numbers which we want
+        # to be larger must have their signs flipped
+        essential_score_tuple = (
+            # Sum of normalized MHC binding affinities of subsequences
             -self.mutant_epitope_score,
-            -self.mutant_protein_fragment.n_alt_reads,
+            # Number of reads supporting the variant
+            -self.mutant_protein_fragment.n_alt_reads
+        )
+        manufacturability_score_tuple = tuple(self.manufacturability_scores)
+        extra_score_tuple = (
+            # Number of reads supporting the particular protein sequence
+            # sequence we're using for this vaccine peptide. Currently
+            # all vaccine peptides are drawn from the same larger sequence
+            # so this score shouldn't change.
             -self.mutant_protein_fragment.n_alt_reads_supporting_protein_sequence,
+            # Minimize the sum of non-mutant MHC binding scores
             self.wildtype_epitope_score,
+            # All else being equal, we prefer to maximize the number of
+            # mutant amino acids
             -self.mutant_protein_fragment.n_mutant_amino_acids,
+            # If nothing else can serve as a tie break then try to center
+            # the mutation in the vaccine peptide.
             -self.mutant_protein_fragment.mutation_distance_from_edge
         )
+        return (
+            essential_score_tuple +
+            manufacturability_score_tuple +
```

Comment:
Thanks for replying @JPFinnigan. We've talked about the wisdom of baking in manufacturability when we might want to eventually use other peptide synthesis methods or even mRNA vaccines. I think the right approach is to give the user options for *which* manufacturability scores to use since there will probably (at the very least) always be some undesirable N and C terminal residues. So, think of this as a first version which will get generalized once we want to use some other kind of peptide synthesis and/or vaccine.